ttt/board.rb

```diff
@@ -17,15 +17,27 @@ def current_state
     @array
   end
 
+  def update_board_state(board_array)
+    @array = board_array
+  end
+
   def make_move(position, symbol)
     @array[position] = symbol
   end
   
-  def available_spots
+  def number_of_empty_positions
     @array.select { |index| index.is_a? Integer }
           .length
   end
 
+  def find_all_empty_positions
+    @array.select { |cell| cell.is_a? Integer }
+        .map! { |value| value - 1 }
```

Comment:
Just in case you haven't come across the difference between `map` and `map!`. `map!` does the usual mapping and also changes the original array.

Reply:
changed to use `map` since I want the original array to not change